docs/tenancy/1.x/hooks-custom.md

```diff
@@ -0,0 +1,242 @@
+---
+title: Custom lifecycle hooks
+icon: fal fa-gear
+excerpt: >
+    How to create custom lifecycle hooks
+tags:
+    - hooks
+    - tenant
+---
+
+# Lifecycle Hooks: Custom Hooks
+
+- [Overview](#overview)
+- [Base](#base)
+- [Implementation](#implementation)
+  - [General](#general)
+  - [Differences](#differences)
+- [Registration](#registration)
+- [Advance](#advance)
+  - [ConfiguresHook Event](#configureshook-event)
+
+## Overview
+
+**Purpose**
+
+Enable custom functionality that isn't available though existing hooks.
+
+**Requirements**
+
+- Knowledge of how [Lifecycle Hooks work](hooks-general)
+
+
+## Base
+
+You can build your own hook by implementing the `Tenancy\Contracts\LifecycleHook` contract.
+
+There are two abstract Hook classes that make implementation easier:
+
+- `Tenancy\Lifecycle\Hook`
+- `Tenancy\Lifecycle\ConfigurableHook`
+
+The basic difference is that the hook settings are not expected change in a `Hook`,
+but are expected to be fluid in the `ConfigurableHook`.
+
+I.e. A Lifecycle Hook that always run at the same priority on the same queue should use `Hook`,
+but a LifecycleHook that will run on the Tenant's Queue should use `ConfigurableHook`
+
+## Implementation
+
+### General
+
+Implementation only requires you to create a `fire` method that contains the logic for the hook.
+
+```php
+public function fire()
+{
+    echo "Running hook for ".$this->event->tenant->name;
+}
+```
+
+If the hook needs to implement different logic based on the event (I.e. Created vs Deleted) you can check the type of event in the `fire` method.
+
+```php
+public function fire()
+{
+    // Created Event
+    if($this->event instanceof \Tenancy\Tenant\Events\Created)
+    {
+        echo "Running hook for the creation of a tenant";
+        return;
+    }
+    // Deleted Event
+    if($this->event instanceof \Tenancy\Tenant\Events\Deleted)
+    {
+        echo "Running hook for the deletion of a tenant";
+        return;
```

Comment:
Can we replace all these echo's with dumps if we want to keep them in like this?